stm32: clock_control: Add test suite for stm32h7

erwango opened the pull request:
Similarly to what was done for U5 in https://github.com/zephyrproject-rtos/zephyr/pull/40731,
add some tests dedicated to H7 clock_control driver configuration

erwango committed:
tests/drivers/clock_control: stm32: Move u5 tests in dedicated folder

Before adding more tests, organize things a little.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32u5: Use a clear_clocks overlay

Instead of relying on existing board clock configuration,
use a clear_clocks.overlay file to first reset the clock
configuration to the default .dtsi state, then apply a
new configuration.
This method should be more robust when trying to use on more
boards and has the benefit to provide correct configuration
examples.

This relies on the fact that overlays are applied in the order
they are provided in DTC_OVERLAY_FILE CMake variable.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32u5: Add a pll_msis_160 overlay

Instead of relying on default board configuration,
add a specific test for this config.

Additionally rename existing pll_msi_80 to pll_msis_160.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32u5: Factorize tests in yaml file

To ease maintenance, add a common section.
It appears that using DTC_OVERLAY_FILE in the common section
preserves the required overlay order.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
tests/drivers/clock_control: stm32: Add test suite for H7 series

Add clock_control test suite for H7 boards

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
This PR requires https://github.com/zephyrproject-rtos/zephyr/pull/41304 to pass the tests

FRASTM commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt:
```diff
@@ -3,7 +3,7 @@
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
-project(stm32_clock_configuration)
+project(stm32_clock_configuration_h5)
```
Is the `project(stm32_clock_configuration_h5) `or project(stm32_clock_configuration_**u**5)

erwango replied:
Right, this is now fixed. Txs

ABOSTM replied:
Still need to be changed to stm32_clock_configuration_u5

FRASTM reviewed (commented):
LGTM

erwango committed:
dts/arm/st: stm32h7: Remove clk_hse default clock-frequency

HSE clock frequency value depends on target.
It can't be defaulted.
Remove this property to ensure it is set properly for each target.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ABOSTM commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_24.overlay:
```diff
@@ -0,0 +1,29 @@
+
+/*
+ * Warning: This overlay performs configuration from clean sheet.
+ * It is assumed that is is applied after clear_clocks.overlay file.
```
typo: "it is"

ABOSTM commented on tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_48.overlay:
```diff
@@ -0,0 +1,29 @@
+
+/*
+ * Warning: This overlay performs configuration from clean sheet.
+ * It is assumed that is is applied after clear_clocks.overlay file.
```
dito

ABOSTM commented on dts/arm/st/h7/stm32h7.dtsi:
```diff
@@ -46,9 +46,6 @@
 		clk_hse: clk-hse {
 			#clock-cells = <0>;
 			compatible = "st,stm32-hse-clock";
-			/* H7 clock driver may not always need this value */
```
As per former comment, I understand that the property is required by binding, even if board doesn't have HSE at all.
By removing this default value, it becomes necessary for board without HSE, to declare this property.

erwango replied:
This property is required by binding only if node is enabled (it is disabled by default).

erwango commented:
@ABOSTM, thanks for the review, I think all is fixed now
